src/configure/clients/TemplateServiceClientFactory.ts

```diff
@@ -1,28 +1,43 @@
 import { ServiceClientCredentials, TokenCredentials } from "ms-rest";
 import { RemoteServiceUrlHelper, ServiceFramework } from "../helper/remoteServiceUrlHelper";
+import { Messages } from '../resources/messages';
 import { TemplateServiceClient } from "./github/TemplateServiceClient";
 import { ITemplateServiceClient } from "./ITemplateServiceClient";
 const UserAgent = "deploy-to-azure-vscode";
 
 export class TemplateServiceClientFactory {
 
-    public static async getClient(credentials?: ServiceClientCredentials, githubPatToken?: string): Promise<ITemplateServiceClient> {
+    public static async getClient(): Promise<ITemplateServiceClient> {
         if (!!this.client) {
             return this.client;
         }
+
+        if (!this.githubPatToken || !this.credentials) {
```

Comment:
Even though githubPat is not used for VSSF service framework, i have kept this githubPAT check condition because template service client will be only used  for github repos and there we will have a valid PAT and for azure-repo we use local templates